cmd/livepeer/livepeer.go

```diff
@@ -695,6 +695,8 @@ func main() {
 				glog.Fatal("Requires a path to the verifier shared volume when local storage is in use; use -verifierPath, S3 or GCS")
 			}
 			verification.VerifierPath = *verifierPath
+		} else if *network != "offchain" {
```

Comment:
We already check `*verifierUrl != ""` ~15 lines above that.  I'm not sure what we gain from this.

...or will that be clearer after #1368 is merged?

Reply:
I'm suggesting to move the `*verifierUrl` check. If you don't nest the `*verifierUrl` check within the on-chain check then when a user specifies a value for the `-verifierUrl` flag *and* runs the node in on-chain mode then the node will hit a problem when it tries to verify a non-existent signature when working with an off-chain O. Furthermore, verification policies should only be necessary when the node is in on-chain mode.

Reply:
> Furthermore, verification policies should only be necessary when the node is in on-chain mode.

At least for me, all the verification development happened off-chain. Since we're incorporating the sig check in the other PR, I can see how that could make things a bit complicated: an offchain verifier (say, for development / testing) would run into sig checking errors without further onchain checks within the verifier. But I'm not sure how common that use case will actually be going forward. So simply doing the check as @yondonfu suggested seems OK. Otherwise if we want to preserve the ability to specify the Epic verifier offchain, then the current structure should probably be preserved, such as if there might ever be a desire to isolate verification testing from more general onchain testing.

I'm indifferent though, since I don't really have further plans to work on this area of the code.

Reply:
got it.  following yondon's suggestion here https://github.com/livepeer/go-livepeer/pull/1368/commits/9946cc1907bf9e56c35580ed787b3f980bfc5433 in #1368

Reply:
I think this suggested change might be causing the hang in circle CI from PR #1368 - https://circleci.com/gh/livepeer/go-livepeer/4596?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link ...

Reply:
> I think this suggested change might be causing the hang in circle CI from PR

Looks like the failure is due to a flaky test. Tracking in #1380 

> Otherwise if we want to preserve the ability to specify the Epic verifier offchain, then the current structure should probably be preserved, such as if there might ever be a desire to isolate verification testing from more general onchain testing.

Ah yeah good point. Ra has mainly been doing verification testing in off-chain mode since it is much easier to setup so I think its a good idea to still allow the Epic verifier to be used in off-chain mode for now. I'm ok keeping this code as-is then and then adding an on-chain check prior to the sig verification in #1368